### PR TITLE
Allow filtering on specific label(s)

### PR DESCRIPTION
**Pull request by mancdaz**
Add a new flag to allow only issues that have certain labels to be
included in the written output to csv.

### CODE REVIEW

File: gh2/csv.py [hunk 1]
```diff
@@ -52,6 +52,14 @@ def make_parser():
         action='store_true', default=False,
     )
     args.add_argument(
+        '--filter-labels',
```
Comment (git-harry) [inline gh2/csv.py]: Given that this need to be specified multiple times to filter by multiple labels, I'd be inclined to call this '--filter-label'.

File: gh2/csv.py [hunk 2]
```diff
@@ -167,13 +175,16 @@ def normalize_sequential_dates(issue_list):
...
 def write_rows(filename, headers, fields, issues, date_format, include_prs,
-               skip_normalization, additional_labels):
+               skip_normalization, additional_labels, filter_labels=None):
     with open(filename, 'w') as fd:
         writer = csv.writer(fd)
         writer.writerow(headers)
         for issue in issues:
             if not include_prs and is_pull_request(issue):
                 continue
+            issue_labels = [l.name for l in issue.labels()]
```
Comment (git-harry) [inline gh2/csv.py]: Given this is used as a set, this could be rewritten as:
issue_labels = {l.name for l in issue.labels()}

File: gh2/csv.py [hunk 3]
```diff
@@ -167,13 +175,16 @@ def normalize_sequential_dates(issue_list):
...
+            if filter_labels and not set(filter_labels).issubset(issue_labels):
```
Comment (git-harry) [inline gh2/csv.py]: It would be better to create filter_labels as a set outside of the for-loop so it's not being rebuilt each time.

**Review by git-harry — APPROVED** [review]
Seems to work as expected. A few nits but nothing to justify reworking.

With the use of '--include-labels' this filtering can be done on the csv file e.g. in Excel or the agile tool. Is this patch addressing a problem with that method?

**Comment by mancdaz** [comment]
@git-harry https://waffle.io/rcbops/u-suk-dev/cards/589b80d901977328000f6ad5 is a requirement for producing csv output that is scoped only to certain labels (rather than all labels in file and filtering in the tool).

**Comment by mancdaz** [comment]
thanks @git-harry changes made as suggested. Merging.

**Pull request merged**